### PR TITLE
Adding Test scenarios

### PR DESCRIPTION
**Pull request by sashikamw**
- JDK - 1.8
- OS - Ubuntu

### CODE REVIEW

File: product-scenarios/1-user-self-registration-to-an-application/1.1-user-registration-with-web-application-itself/1.1.1-provision-user-using-SCIM2/src/test/java/org/wso2/identity/scenarios/test/scim2/ProvisionUserSCIM2.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
+package org.wso2.identity.scenarios.test.scim2;
```
Comment (kasunbg) [inline product-scenarios/1-user-self-registration-to-an-application/1.1-user-registration-with-web-application-itself/1.1.1-provision-user-using-SCIM2/src/test/java/org/wso2/identity/scenarios/test/scim2/ProvisionUserSCIM2.java]: missing license.

Comment (sashikamw) [reply product-scenarios/1-user-self-registration-to-an-application/1.1-user-registration-with-web-application-itself/1.1.1-provision-user-using-SCIM2/src/test/java/org/wso2/identity/scenarios/test/scim2/ProvisionUserSCIM2.java]: Adding the license

File: product-scenarios/1-user-self-registration-to-an-application/1.1-user-registration-with-web-application-itself/1.1.1-provision-user-using-SCIM2/src/test/java/org/wso2/identity/scenarios/test/scim2/ProvisionUserSCIM2TestCase.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+    private String getIdentityHTTPSEP() {
+
+        String bucketLocation = System.getenv("DATA_BUCKET_LOCATION");
+        String url = null;
+        log.info("Data Bucket location is set : " + bucketLocation);
+
+        Properties prop = new Properties();
+        //InputStream input = null;
+        try (InputStream input = new FileInputStream(bucketLocation + "/infrastructure.properties")) {
```
Comment (IsuraD) [inline product-scenarios/1-user-self-registration-to-an-application/1.1-user-registration-with-web-application-itself/1.1.1-provision-user-using-SCIM2/src/test/java/org/wso2/identity/scenarios/test/scim2/ProvisionUserSCIM2TestCase.java]: use constant

File: product-scenarios/1-user-self-registration-to-an-application/1.1-user-registration-with-web-application-itself/1.1.1-provision-user-using-SCIM2/src/test/java/org/wso2/identity/scenarios/test/scim2/ProvisionUserSCIM2TestCase.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+            prop.load(input);
+            url = prop.getProperty("ISHttpsUrl");
```
Comment (IsuraD) [inline product-scenarios/1-user-self-registration-to-an-application/1.1-user-registration-with-web-application-itself/1.1.1-provision-user-using-SCIM2/src/test/java/org/wso2/identity/scenarios/test/scim2/ProvisionUserSCIM2TestCase.java]: use constanat

File: product-scenarios/1-user-self-registration-to-an-application/1.1-user-registration-with-web-application-itself/1.1.1-provision-user-using-SCIM2/src/test/java/org/wso2/identity/scenarios/test/scim2/ProvisionUserSCIM2TestCase.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        } catch (IOException ex) {
+            ex.printStackTrace();
```
Comment (IsuraD) [inline product-scenarios/1-user-self-registration-to-an-application/1.1-user-registration-with-web-application-itself/1.1.1-provision-user-using-SCIM2/src/test/java/org/wso2/identity/scenarios/test/scim2/ProvisionUserSCIM2TestCase.java]: log the error. Do not use printStackTrace

File: product-scenarios/1-user-self-registration-to-an-application/1.1-user-registration-with-web-application-itself/1.1.1-provision-user-using-SCIM2/src/test/java/org/wso2/identity/scenarios/test/scim2/ProvisionUserSCIM2TestCase.java [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+        }
+        if (url == null) {
+            url = "https://localhost:9443";
```
Comment (IsuraD) [inline product-scenarios/1-user-self-registration-to-an-application/1.1-user-registration-with-web-application-itself/1.1.1-provision-user-using-SCIM2/src/test/java/org/wso2/identity/scenarios/test/scim2/ProvisionUserSCIM2TestCase.java]: use constants to build the default IS url

Comment (sashikamw) [reply product-scenarios/1-user-self-registration-to-an-application/1.1-user-registration-with-web-application-itself/1.1.1-provision-user-using-SCIM2/src/test/java/org/wso2/identity/scenarios/test/scim2/ProvisionUserSCIM2TestCase.java]: Updated using the base class

File: product-scenarios/1-user-self-registration-to-an-application/1.1-user-registration-with-web-application-itself/1.1.1-provision-user-using-SCIM2/pom.xml [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <groupId>org.wso2.is</groupId>
+        <artifactId>user-self-registration-using-webapplication-test</artifactId>
+        <version>5.7.0</version>
```
Comment (janakamarasena) [inline product-scenarios/1-user-self-registration-to-an-application/1.1-user-registration-with-web-application-itself/1.1.1-provision-user-using-SCIM2/pom.xml]: Should we change the pom version to match the version in the master(5.8.0-m8-SNAPSHOT)?

**Pull request merged**